pilot/test/util/kubernetes.go

```diff
@@ -38,28 +38,60 @@ const (
 func CreateNamespace(cl kubernetes.Interface) (string, error) {
 	return CreateNamespaceWithPrefix(cl, "istio-test-", false)
 }
+// CreateNamespaceWithPrefixorName creates a fresh namespace with either the provided prefix or the provided name
+func CreateNamespaceWithPrefixorName(cl kubernetes.Interface, prefix string, inject bool, namespace string) (string, error) {
 
-// CreateNamespaceWithPrefix creates a fresh namespace with the given prefix
-func CreateNamespaceWithPrefix(cl kubernetes.Interface, prefix string, inject bool) (string, error) {
 	injectionValue := "disabled"
 	if inject {
 		injectionValue = "enabled"
 	}
-	ns, err := cl.CoreV1().Namespaces().Create(&v1.Namespace{
-		ObjectMeta: meta_v1.ObjectMeta{
-			GenerateName: prefix,
-			Labels: map[string]string{
-				"istio-injection": injectionValue,
+	var err error
+	var ns *v1.Namespace
+	if namespace == "" {
+		ns, err = cl.CoreV1().Namespaces().Create(&v1.Namespace{
+			ObjectMeta: meta_v1.ObjectMeta{
+				GenerateName: prefix,
+				Labels: map[string]string{
+					"istio-injection": injectionValue,
+				},
+			},
+		})
+	} else {
+		ns, err = cl.CoreV1().Namespaces().Create(&v1.Namespace{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Name: namespace,
+				Labels: map[string]string{
+					"istio-injection": injectionValue,
+				},
 			},
-		},
```

Comment:
The only difference between the if and else code is "GenerateName: prefix" and "Name: namespace". 
Anyone know how to update to pass that in as an object (code below is wrong/guess)? That would simplify this code block to:
if namespace == "" {
   keyName = "GenerateName:"
   valueName = prefix
} else {
   keyName = "Name:"
   valueName = namespace
}
ns, err := c1.....Create(...)